### DISCUSSION TITLE
Hidden Boolean

### DISCUSSION THREAD

**Discussion by n0pe-01** [Budibase/budibase#13775] [Help] [answered]
Hi

I'm fairly new to Budibase and I'm trying setup a basic onboard form with some API requests in the background. 

I have setup a Multistep Form block based off a table that has a boolean column (UserCreated). This column isn't displayed on the form because it will be used later with my API request. 

When i save a new form the column is empty instead False as i was expecting.  I've set the default value to false but still if i don't add the column into the block it wont set a value. I could work around this if the column appeared in the POST request but it doesn't.

I have tried adding a update field action to the save button but still i can't get the field updated when saved. 

Does anyone have any idea on how i can set the default value without having to add the column into the form block?

**Form and Column**
![Form](https://github.com/Budibase/budibase/assets/113320443/ca76f4dc-b301-4ef3-a375-f22302023fd5)

**After saved**
![Row](https://github.com/Budibase/budibase/assets/113320443/92fa9d5e-2f99-4c27-ab3b-dc6a459309bc)

**API Response** 
![API](https://github.com/Budibase/budibase/assets/113320443/5c99c299-fa80-4f94-98b6-3bf363a0fdfd)

**Comment by mikesealey** [comment, marked as answer]
Hey @n0pe-01 and welcome to Budibase!

_"Does anyone have any idea on how i can set the default value without having to add the column into the form block?"_

This is totally achievable, but requires a slightly different approach. You're using a `Multi Step Form Block`, which is a really good one-size-fits-all approach that works in _most_ use-cases, but for what you're looking to do, it might be better to take a _slightly_ different approach using a `Form` component instead.

![image](https://github.com/Budibase/budibase/assets/39269380/8f217398-bf72-48b7-bbc3-531f5480a179)

You'll need to build your form manually using components.

- Start by adding a Form component
- Next, click to add another component and search for "Field". This will give you all of the options to collect a user-input.
- Keep building your form out to collect all of the data you want from the user.
- Remember to add to the `Field` setting in each input field
![image](https://github.com/Budibase/budibase/assets/39269380/61da0346-7e79-42c3-a821-6926441c63af)

Once your form looks all set to go, add a `Button` component as a child of your form component
![image](https://github.com/Budibase/budibase/assets/39269380/3c734ff5-ee31-41b6-bfb3-42de82bf3c38)

- Set an on-click action of your button to `Save Row`
- 
![image](https://github.com/Budibase/budibase/assets/39269380/6e0d4b24-d5d3-4e3b-bf63-738139f28672)

- Here, you should select the form you've just created as the Datasource, and the table you wish to create a row in.
- Click Add Column, and pick out the columns you want to save, setting the values as those captured from the form
- **Important** Add an additional column that isn't mentioned in your form, but is a column in your table - you can now set this value as your choose. In your instance it would be something along the lines of **Column:** `UserCreated` **Value:** `False`

Hopefully this should get you on your way. If you haven't already, you could join the [Budibase Discord Server](https://discord.com/invite/ZepTmGbtfF) 

**PS:** You might be able to achieve a similar result using Budibase Auto-Columns, but if the above approach works for you, that's fine too. Here's the documentation you can read about it: https://docs.budibase.com/docs/budibasedb#auto-columns

**Reply by n0pe-01** [reply]
Hi Mike

Thanks for your reply. I've followed you suggestion but i must be missing something. 

With the **UserCreated** column added to save row action the form doesn't save. If i remove that option it saves. 

As a test i created a new text column and set that column to update to a random text value and that worked perfectly.  So now i believe it doesn't like the value of "false" when updating a boolean.  Is there some sort of other function i should be using or should simply "true" or "false" work?

I could just use a text field instead of a boolean but i would prefer to use the proper column type. 


![Form](https://github.com/Budibase/budibase/assets/113320443/7157f59f-ed75-4968-969c-42122950b045)

**Comment by mikesealey** [comment]
Easy fix: don't capitalise `true` or `false`. 

One step I also forgot to add to the first response: You might also want a `Clear Form` action after you've saved the row.

**Reply by n0pe-01** [reply]
**slaps forehead**!
So obvious thank you.
